parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -355,10 +355,11 @@ if ! [[ ${API_SERVER_NAME} =~ ^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
             VALIDATION_ERR=$ERR_K8S_API_SERVER_DNS_LOOKUP_FAIL
         fi
     else
-        logs_to_events "AKS.CSE.apiserverNC" "retrycmd_if_failure ${API_SERVER_CONN_RETRIES} 1 10 nc -vz ${API_SERVER_NAME} 443" || time nc -vz ${API_SERVER_NAME} 443 || VALIDATION_ERR=$ERR_K8S_API_SERVER_CONN_FAIL
+        logs_to_events "AKS.CSE.apiserverCurl" "retrycmd_if_failure ${API_SERVER_CONN_RETRIES} 1 10 curl -v --insecure https://${API_SERVER_NAME}:443" || time curl -v --insecure "https://${API_SERVER_NAME}:443" || VALIDATION_ERR=$ERR_K8S_API_SERVER_CONN_FAIL
```

Comment:
do we actually want the `--insecure` option here?

Reply:
I guess it will still use TLS but just won't verify the API server's cert

Reply:
It'll return self-signed cert error without it, I think.

Reply:
Correct. Since it's self signed. You need  -k to have a 0 return code